tools/sof-tplgreader.py

```diff
@@ -16,43 +16,40 @@ def __init__(self):
     def __comp_pipeline(self, pipeline):
         return pipeline['id']
 
-    # fork & split from TplgFormatter
-    def _fmt2str(self, cap):
-        fmt = cap['formats']
-        if fmt & (1 << 2) != 0:
-            return "S16_LE"
-        elif fmt & (1 << 6) != 0:
-            return "S24_LE"
-        elif fmt & (1 << 10) != 0:
-            return "S32_LE"
-        elif fmt & (1 << 14) != 0:
-            return "FLOAT"
-        else: # without match
-            return ""
+    # supported formats of playback pipeline in formats[0]
+    # supported formats of capture pipeline in formats[1]
+    def _fmt2str(self, formats, pcm_type):
+        fmts_str = ""
+        if pcm_type == "capture":
+            for item in formats[1]:
+                fmts_str = fmts_str + " " + item
+        else: # for playback and both pcm, use playback formats as default
+            for item in formats[0]:
+                fmts_str = fmts_str + " " + item
+        return fmts_str.strip()
 
     def _key2str(self, cap, key):
         return cap["%s_min" % (key)], cap["%s_max" % (key)]
 
     # fork & split from TplgFormatter
     def loadFile(self, filename, sofcard=0):
         tplg_parser = TplgParser()
-        for item in tplg_parser.parse(filename)[:-1]:
+        parsed_tplg = tplg_parser.parse(filename)
+        tplg_formatter = TplgFormatter(parsed_tplg)
+        # ignore the last element, for it is tplg name
+        for item in parsed_tplg[:-1]:
             if "pcm" not in item:
                 continue
             for pcm in item['pcm']:
                 pipeline_dict = {}
                 pipeline_dict['pcm'] = pcm["pcm_name"]
                 pipeline_dict['id'] = str(pcm["pcm_id"])
-                if pcm['playback'] == 1 and pcm['capture'] == 1:
-                    pipeline_dict['type'] = 'both'
-                elif pcm['playback'] == 1:
-                    pipeline_dict['type'] = 'playback'
-                elif pcm['capture'] == 1:
-                    pipeline_dict['type'] = 'capture'
-                else: # error pcm ??
-                    return 1
+                pipeline_dict['type'] = tplg_formatter._get_pcm_type(pcm)
```

Comment:
remove code without handle 'none', I split this code from TplgFormatter,  
because I think the none is ok for tplgreader, but for the test-case without process to handle the 'none'  
so here will catch the function return